src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -2518,23 +2514,34 @@ namespace winrt::TerminalApp::implementation
         // and wait on it hence the locking mechanism.
         if (Dispatcher().HasThreadAccess())
         {
-            // TODO: GH 9458 will give us more context so we can try to choose a better profile.
-            auto hr = _OpenNewTab(nullptr, connection);
-
-            // Request a summon of this window to the foreground
-            _SummonWindowRequestedHandlers(*this, nullptr);
+            try
+            {
+                NewTerminalArgs newTerminalArgs{};
+                // TODO GH#XXXXX: When we pass the actual commandline (or originating application), the
+                // settings model can choose the right settings based on command matching, or synthesize
+                // a profile from the registry/link settings (TODO GH#9458).
+                // TODO GH#9458: Get and pass the LNK/EXE filenames.
+                newTerminalArgs.Commandline(L"default-terminal-invocation-placeholder");
```

Comment:
I feel worst about this. It was either "use the new commandline detection in `GetProfileForArgs`" or "copy the feature flag detection code over to TerminalPage as well" 😦  (since it needs to work properly for all types of invocation.)

Reply:
I'm confused. Why do we need to pass in `"default-terminal-invocation-placeholder"`? `GetProfileForArgs()` checks for a profile index and a profile name. Why can't we just pass in `nullptr` directly?

Reply:
This is actually *the* line that causes it to choose the base profile instead of the default one.

Reply:
Discussed offline. I would add a short comment saying something like `// Passing in a commandline forces GetProfileForArgs to use Base Layer instead of Default Profile`. It's a bit confusing since this affects something one or two function calls down.

Other than that, I now understand your pain hahaha